R/viz.R

```diff
@@ -4367,37 +4383,46 @@ fancy_sv_sizedens = function(this_sample,
 #' @export
 #'
 #' @examples
-#' sample_table = data.frame(sample_id  = c("09-33003T", "05-18426T", "SP59282"))
-#' fancy_qc_plot(these_samples = sample_table, seq_type = "genome")
+#' all_fl = dplyr::filter(this_meta, pathology == "FL") %>%
+#'   dplyr::select(sample_id) %>%
+#'   arrange(sample_id) %>%
+#'   as.data.frame()
 #'
-fancy_qc_alignment_plot = function(these_samples,
-                         metadata,
-                         comparison_data = NULL,
-                         seq_type = "genome",
-                         add_mean = TRUE,
-                         limit_cohort = NULL,
-                         limit_pathology = NULL,
-                         this_color_palette = c("TotalReads" = "#3D405B",
-                                                "TotalUniquelyMapped" = "#81B29A",
-                                                "TotalDuplicatedreads" = "#E07A5F"),
-                         plot_sub = ""){
-
-  if(missing(these_samples) && missing(metadata)){
-    metadata = get_gambl_metadata()
-  }
+#' all_bl = dplyr::filter(this_meta, pathology == "BL") %>%
+#'   dplyr::select(sample_id) %>%
+#'   arrange(sample_id) %>%
+#'   as.data.frame()
+#'
+#' plot = fancy_alignment_plot(these_samples = all_fl, seq_type = "genome", comparison_group = all_bl, plot_sub = "all FL cases (BL comparison group)")
+#'
+fancy_alignment_plot = function(these_samples,
+                                metadata,
+                                comparison_group,
+                                seq_type = "genome",
+                                add_mean = TRUE,
+                                filter_cohort,
+                                filter_pathology,
+                                this_color_palette = c("TotalReads" = "#3D405B", "TotalUniquelyMapped" = "#81B29A", "TotalDuplicatedreads" = "#E07A5F"),
+                                plot_sub = ""){
 
-  if(missing(these_samples) && !is.null(limit_cohort)){
-    these_samples = dplyr::filter(metadata, cohort == limit_cohort) %>%
-      dplyr::select(sample_id) %>%
-      arrange(sample_id) %>%
-      as.data.frame()
+  #get gambl metadata (if not supplied)
+  if(missing(metadata)){
+    this_meta = get_gambl_metadata()
+  }else{
+    this_meta = metadata
   }
 
-  if(missing(these_samples) && !is.null(limit_pathology)){
-    these_samples = dplyr::filter(metadata, pathology == limit_pathology) %>%
-      dplyr::select(sample_id) %>%
-      arrange(sample_id) %>%
-      as.data.frame()
+  #filter metadata on selected cohort (if filter_cohort is called)
+  if(missing(these_samples)){
```

Comment:
This will not work as expected if someone supplies BOTH filter_cohort AND filter_pathology, will it?

Reply:
No, it will not. The way I designed this function was so that the user can define one or the other. But I can see how it could be useful to further subset your data based on both conditions. I will address this.